scripts/cli/src/util/checkBridgeBalance.ts

```diff
@@ -0,0 +1,164 @@
+		return timelock.toNumber() * 6;
+	}
+
+	function currentTimeToSeconds() {
+		const currentDateTime = new Date();
+		const resultInSeconds = currentDateTime.getTime() / 1000;
+		return resultInSeconds;
+	}
+
+	function hoursToSeconds(hours: number) {
+		let seconds = hours * 3600;
+		return seconds;
+	}
+
+	function checksUser(
+		userMap: Map<string, userInfo>,
+		newPolyx: any,
+		offenderAddress: any,
+		blockNumber: number
+	) {
+		const user = userMap.get(offenderAddress)!;
+		const timeDifference = currentTimeToSeconds() - user.lastTimeBridged;
+
+		//check timeDifference to make sure its not less than 16mins
+		// if it is then no modifications should be done as
+		// seperate events have the same information
+		if (timeDifference > hoursToSeconds(0.25)) {
+			const newBalance = user.polyx + newPolyx;
+
+			checkPolyxLimit(newPolyx, offenderAddress, blockNumber);
+
+			if (
+				newBalance > BRIDGE_POLY_TOTAL &&
+				timeDifference <= BRIDGE_TIME_PERIOD
+			) {
+				console.log(`Error: Total POLYX Bridge limit exceeded.`);
+				console.log(
+					"Polyx Total Bridge Limit Equivocation: ",
+					offenderAddress,
+					blockNumber
+				);
+				prom.polyxTotalBridgeLimitEquivocations.inc({ offenderAddress });
+				// resets polyx to zero to check for new errors that break polyx limit
+				userMap.set(offenderAddress, {
+					polyx: 0,
+					lastTimeBridged: currentTimeToSeconds(),
+				});
+			} else {
+				console.log("Updates user data.");
+				userMap.set(offenderAddress, {
+					polyx: newBalance,
+					lastTimeBridged: currentTimeToSeconds(),
+				});
+			}
+		}
+	}
+
+	function checkPolyxLimit(
+		newPolyx: number,
+		offenderAddress: string,
+		blockNumber: number
+	) {
+		if (newPolyx > BRIDGE_LIMIT) {
+			console.log(`Error: Transaction POLYX limit exceeded.`);
+			console.log(
+				"Polyx Bridge Limit Equivocation: ",
+				offenderAddress,
+				blockNumber
+			);
+			prom.polyxBridgeLimitEquivocations.inc({ offenderAddress });
+		}
+	}
+
+	// Subscribe to the new headers on-chain. The callback is fired when new headers
+	// are found, the call itself returns a promise with a subscription that can be
+	// used to unsubscribe from the newHead subscription
+	const unsubscribe = await api.rpc.chain.subscribeNewHeads((header) => {
+		console.log(`Chain is at block: #${header.number}`);
+
+		// Subscribe to system events via storage
+		api.query.system.events((events) => {
+			console.log(`\nReceived ${events.length} events:`);
+
+			events
+				.map((record) => {
+					return record.event;
+				})
+				.filter((event) => {
+					return (
+						event.section === "bridge" && event.method === "BridgeTxScheduled"
+					);
+				})
+				.forEach((event) => {
+					const types = event.typeDef;
+
+					event.data
+						.filter((data, index) => types[index].type === "BridgeTx")
+						.forEach((data) => {
+							const txData = JSON.parse(data.toString());
+							const newPolyx = txData.value;
```

Comment:
Looks like we forgot to sync the `polymesh_schema.json` file for that field name.  The JS Api uses the schema for the types.  The `SCALE` encoding doesn't care about type/field names.

Reply:
Good question, the data that is outputted is in this format {"nonce":0,"recipient":"some addr","value": 100,"tx_hash":"hash"}

Reply:
@JMoore96 Lets bring it up in the stand-up today.  It might be too late to update `polymesh_schema.json` for the mainnet release.  Updating that will have an effect on the SDK team.  @adamdossa

Reply:
@JMoore96 The schema changes have been merged to `develop`.  Once you sync this branch with `develop` the field name should be `amount`.